src/tx/builder/index.ts

```diff
@@ -0,0 +1,743 @@
+ */
+export function unpackRawTx<Tx extends TxParams> (
+  binary: Uint8Array | NestedUint8Array,
+  schema: TxField[]): Tx {
+  return schema
+    .reduce<any>(
+    (
+      acc,
+      [key, fieldType, prefix],
+      index
+    ) => Object.assign(acc, { [key]: deserializeField(binary[index], fieldType, prefix) }),
+    {}
+  )
+}
+
+/**
+ * Get transaction serialization/deserialization schema
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder
+ * @param {{ vsn: String,fieldIndex: string | number, isDeserialized: Boolean }}
+ * @throws {Error} Schema not found error
+ * @return Schema
+ */
+const getSchema = ({ vsn, objId, type }: {
+  vsn: number | string
+  objId?: string
+  type?: string
+}): [schema: TxField[], tag:string | number] => {
+  const isDeserialize = Boolean(objId)
+  const firstKey = isDeserialize ? objId : type
+  const schema = isDeserialize ? TX_DESERIALIZATION_SCHEMA : TX_SERIALIZATION_SCHEMA
+
+  if (firstKey != null && schema[firstKey] != null &&
+    schema[firstKey][vsn] != null) {
+    return schema[firstKey][vsn]
+  } else {
+    if (firstKey != null && (schema[firstKey] == null)) {
+      throw new SchemaNotFoundError(`Transaction ${isDeserialize ? 'deserialization' : 'serialization'} not implemented for ${isDeserialize ? 'tag ' + (objId ?? '') : (type ?? '')}`)
+    }
+    throw new SchemaNotFoundError(`Transaction ${isDeserialize ? 'deserialization' : 'serialization'} not implemented for ${isDeserialize ? 'tag ' + (objId ?? '') : (type ?? '')} version ${vsn}`)
+  }
+}
+
+export interface TxHash<Tx extends TxParams> {
+  tx: EncodedData<'tx'>
+  rlpEncoded: Buffer
+  binary: Uint8Array | NestedUint8Array
+  txObject: Tx
+}
+/**
+ * Build transaction hash
+ * @function
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder
+ * @param params Object with tx params
+ * @param type Transaction type
+ * @param [options={}] options
+ * @param [options.excludeKeys] Array of keys to exclude for validation and build
+ * @param [options.prefix] Prefix of transaction
+ * @throws {Error} Validation error
+ * @returns object
+ * @returns object.tx Base64Check transaction hash with 'tx_' prefix
+ * @returns object.rlpEncoded rlp encoded transaction
+ * @returns object.binary binary transaction
+ */
+export function buildTx<Tx extends TxParams> (
+  params: any,
+  type: string,
+  { excludeKeys = [], prefix = 'tx', vsn = VSN, denomination = AE_AMOUNT_FORMATS.AETTOS } = {}
+): TxHash<Tx> {
+  const [schema, tag] = getSchema({ type, vsn })
+  const binary = buildRawTx(
+    { ...params, VSN: vsn, tag },
+    schema,
+    { excludeKeys, denomination: ('denomination' in params ? params.denomination : undefined) ?? denomination }
+  ).filter((e?: Buffer) => e != null)
+
+  const rlpEncoded = Buffer.from(rlpEncode(binary))
+  const tx: EncodedData<'tx'> = encode<'tx'>(rlpEncoded, prefix as 'tx')
+
+  return { tx, rlpEncoded, binary, txObject: unpackRawTx<Tx>(binary, schema) }
+}
+
+export interface TxHashUnpacked<Tx extends TxParams> {
```

Comment:
```suggestion
export interface TxUnpacked<Tx extends TxParams> {
```
hash can't be unpacked, it is one-way thing